q2_diversity/_filter.py

```diff
@@ -12,13 +12,22 @@
 
 def filter_distance_matrix(distance_matrix: skbio.DistanceMatrix,
                            metadata: qiime2.Metadata,
-                           where: str=None) -> skbio.DistanceMatrix:
+                           where: str=None,
+                           exclude_ids: bool=False) -> skbio.DistanceMatrix:
     ids_to_keep = metadata.ids(where=where)
     # NOTE: there is no guaranteed ordering to output distance matrix because
     # `ids_to_keep` is a set, and `DistanceMatrix.filter` uses its iteration
     # order.
     try:
-        return distance_matrix.filter(ids_to_keep, strict=False)
+        filtered = distance_matrix.filter(ids_to_keep, strict=False)
```

Comment:
We can refactor this to avoid computing `filtered` twice, right? We can use the same strategy as in [q2-feature-table](https://github.com/qiime2/q2-feature-table/blob/master/q2_feature_table/_filter.py#L48-L49).

```python
if exclude_ids:
    ids_to_keep = set(distance_matrix.ids) - set(ids_to_keep)
return distance_matrix.filter(ids_to_keep, strict=False)
```